src/core/issues-notes.js

```diff
@@ -147,59 +175,83 @@ function handleIssues($ins, ghIssues, conf) {
   }
 }
 
-async function fetchIssuesFromGithub({ githubAPI }) {
-  const issues = [];
-  const issueNumbers = [...document.querySelectorAll(".issue[data-number]")]
+async function fetchAndStoreGithubIssues(githubAPI) {
+  const specIssues = document.querySelectorAll(".issue[data-number]");
+  if (specIssues.length > MAX_GITHUB_REQUESTS) {
+    const msg =
+      `Your spec contains ${specIssues.length} Github issues, ` +
+      `but GitHub only allows ${MAX_GITHUB_REQUESTS} requests. Some issues might not show up.`;
+    pub("warning", msg);
+  }
+  const issuePromises = [...specIssues]
     .map(elem => Number.parseInt(elem.dataset.number, 10))
-    .filter(number => number);
-  for (const issueNumber of issueNumbers) {
-    let issue; // depends on how fetching goes...
-    try {
+    .filter(issueNumber => issueNumber)
+    .map(async issueNumber => {
       const issueURL = `${githubAPI}/issues/${issueNumber}`;
-      const response = await fetch(issueURL, {
-        // Get back HTML content instead of markdown
-        // See: https://developer.github.com/v3/media/
-        Accept: "application/vnd.github.v3.html+json",
+      const request = new Request(issueURL, {
+        headers: {
+          // Get back HTML content instead of markdown
+          // See: https://developer.github.com/v3/media/
+          Accept: "application/vnd.github.v3.html+json",
+        },
       });
-      if (!response.ok) {
-        switch (response.status) {
-          case 404:
-            throw new Error(
-              "Couldn't find issue on Github. Check if it exist?"
-            );
-          default:
-            throw new Error(
-              "Network error. Github is down? or too many requests?"
-            );
-        }
+      try {
+        const response = await fetchAndCache(request);
+        return processResponse(response, issueNumber);
+      } catch(err){
+        return processResponse(err.response, issueNumber);
```

Comment:
A fetch error won't always contain a `response` field as it can also be a native `TypeError`. I'm starting to think we are better to keep original `fetch` exception model here. (Throwing only when it couldn't even connect)

Reply:
> A fetch error won't always contain a response field as it can also be a native TypeError.

True. 

> I'm starting to think we are better to keep original fetch exception model here. (Throwing only when it couldn't even connect)

Yeah, that might make more sense.